Data/Aeson/Types/Generic.hs

```diff
@@ -201,7 +201,6 @@ instance ( IsRecord      a isRecord
         (builder tagFieldName <>
          B.char7 ':' <>
          builder (constructorTagModifier opts (conName (undefined :: t c a p)))) <>
-        B.char7 ',' <>
```

Comment:
Yes indeed, I moved the comma insertion. Otherwise, the encoding for nullary constructor would be `{"tag": "X",}` (trailing comma). I agree a separate commit would have made this more clear, I'll keep it in mind.